rust/lance-index/src/scalar/inverted/index.rs

```diff
@@ -434,10 +489,15 @@ impl DeepSizeOf for PostingList {
 }
 
 impl PostingList {
-    pub fn new(row_ids: ScalarBuffer<u64>, frequencies: ScalarBuffer<f32>) -> Self {
+    pub fn new(
+        row_ids: ScalarBuffer<u64>,
+        frequencies: ScalarBuffer<f32>,
+        positions: Option<ListArray>,
```

Comment:
Why is this `Option`?  Is it because older versions of the index might not have this data?

Reply:
Yes, I don't want to introduce a breaking change, so make it optional, so that old index can still work, just doesn't support phrase query, the error message would guide users to re-create index